node/sync/src/block_sync.rs

```diff
@@ -1169,12 +1185,9 @@ impl<N: Network> BlockSync<N> {
             let requests = self.requests.read();
             let mut start_height = sync_height + 1;
 
-            loop {
-                if requests.contains_key(&start_height) {
-                    start_height += 1;
-                } else {
-                    break;
-                }
+            // Do not issue requests that already exist or would be obsolete.
+            while requests.contains_key(&start_height) || self.ledger.latest_block_height() >= start_height {
```

Comment:
`Ledger::latest_block_height` involves an `RwLock::read` operation, so it would be beneficial if we moved it above the loop to obtain it just once; sure, the ledger could move in the meantime, but the likelihood is low, because:
1. block time is a lot greater than what this loop could take
2. we're attempting to sync already, so the ledger is most likely not advancing